### ISSUE TITLE
The cmake doesn't generate config file to be included as a 3th party project

### ISSUE THREAD

**Issue by arsdever** [fpagliughi/sockpp#45]
I am developing a library, which uses come 3rd party libraries including this one. For the other libraries, I was able to include them using the following `CMakeLists.txt` code:
```cmake
find_package(package_name REQUIRED)
target_link_libraries(my_target_name PRIVATE package_name::package_name)
```
but for this one, it says the following:

> Could not find a package configuration file provided by "sockpp-static" with any of the following names:
> sockpp-staticConfig.cmake
> sockpp-static-config.cmake

**Comment by arsdever** [comment]
I did some investigation, and found out, that the other libraries are using the following lines in their `CMakeLists.txt` files:

```cmake
export(PACKAGE ${PROJECT_NAME})
```

No

**Comment by varunsh-xilinx** [comment]
@fpagliughi I saw you pulled these changes into the `cmake-refactor` branch. Are you planning on merging it to master in the near future?

**Comment by fpagliughi** [comment]
@varunsh-xilinx  I've been trying to learn more CMake, but haven't been doing a lot of C++ programming lately, so it's been slow. I will go back to testing it and see what I still needed to do to finish, but if you test it, let me know how it works out for you.

**Comment by varunsh-xilinx** [comment]
I'm using the cmake-refactor branch currently and it seems to work fine. Great work on this project btw!

**Comment by stephenaa** [comment]
Using the cmake-refactor branch, what is the correct way of bringing this into another, top-level, CMake project?

**Comment by varunsh-xilinx** [comment]
@stephenaa I'm just installing this project using the cmake-refactor branch. Then, in the other CMake project, I'm using `find_package(sockpp)` and then linking to it as in the code snippet above.

**Comment by fpagliughi** [comment]
This should finally be fixed with a released version - v0.8.0
